client/packages/common/src/localStorage/keys.ts

```diff
@@ -26,6 +29,9 @@ export type LocalStorageRecord = {
   '/auth/error': AuthError | undefined;
   '/pagination/rowsperpage': number;
   '/columns/hidden': Record<string, string[]> | undefined;
-};
+} & Record<
+  `/user/${string}/store/${string}/selectedVariants`,
+  UserSelectedVariants
+>;
```

Comment:
I couldn't add the template string literat to the existing type but had to add it like that...

Reply:
now that's a neat solution.
I've used the stored data to define runtime variants with a static key - the hidden columns for example, there the key is `/columns/hidden` but the value stores the table name

Reply:
Also a good idea! should I follow the same pattern? i.e. have something like:

```ts
selectedUnitVariants: {
  [storeId]: {
    [userId]: UserSelectVairant
  }
}
```

Never the less I should rename `/user/${string}/store/${string}/selectedVariants` to `/user/${string}/store/${string}/selectedUnitVariants`, right?

Reply:
Do we actually assume that two users are using the same browser? Would make sense, but then we should sort other settings by user as well, e.g. `/user/{userId}/columns/hidden`, shouldn't we?

Any objections or preferences to use the format from the PR though? For example, when using a static key and then have nested objects it becomes a bit of a pain to edit them, e.g. when a new user or store is added the parent object need be created as well which is a bit error-prone... Not a big thing but much easier when encoding it in the key...

Reply:
I agree with keeping the format in the PR, and I wonder about extending it out, so that local storage keys can be prefixed with the user id and in some cases the store id too. We could write a getter which would take the current user and fetch all the key value pairs for that user, before returning the requested key. It just means moving around AuthContext and LocalStorage so that we can fetch the user and store from the context, and refactoring GqlContext to access local storage directly, without the auth context.

Reply:
Ok great. I changed the key name to `/user/${string}/store/${string}/selectedunits` thought its more fitting and uses existing lower-case key parts.